### PR TITLE
[MRG+1] Support custom encoding in w3lib.http.basic_auth_header

### LINKED ISSUE #88: allow to customize encoding in w3lib.http.basic_auth_header
See https://github.com/scrapy/scrapy/pull/2530

### PR DESCRIPTION
**Pull request by elacuesta**
Closes #88

### CODE REVIEW

File: tests/test_http.py [hunk 1]
```diff
@@ -13,6 +15,11 @@ def test_basic_auth_header(self):
         # Check url unsafe encoded header
         self.assertEqual(b'Basic c29tZXVzZXI6QDx5dTk-Jm8_UQ==',
             basic_auth_header('someuser', '@<yu9>&o?Q'))
+        # custom encoding
+        self.assertEqual(b'Basic c29t5nVz6HI6c_htZXDkc3M=',
```
Comment (kmike) [inline tests/test_http.py]: Could you please move custom encoding tests to a separate test method?

**Comment by kmike** [comment]
Could you please rebase it, to fix CI? Code looks good to me.

**Comment by elacuesta** [comment]
Updated, thanks Mikhail!

**Comment by kmike** [comment]
So, the next step after merging this PR and releasing a new w3lib version is to bump w3lib version in Scrapy's setup.py and remove some copy-paste from proxy auth handling?

**Comment by elacuesta** [comment]
That sounds great, I can take care of it after this is released.

**Comment by redapple** [comment]
Thanks @elacuesta !

**Pull request merged**